### PR TITLE
iouring sq poll and io poll

### PR DESCRIPTION
**Pull request by lihuiba**

### CODE REVIEW

File: io/iouring-wrapper.cpp [hunk 1]
```diff
@@ -198,6 +241,10 @@ class iouringEngine : public MasterEventEngine, public CascadingEventEngine, pub
             io_uring_sqe_set_data(sqe, &timer_ctx);
         }
 
+        int ret = io_uring_submit(m_ring);
+        if (ret < 0)
+            LOG_ERROR_RETURN(-ret, -1, "iouring: fail to submit when adding interest, ", ERRNO(-ret));
+
```
Comment (beef9999) [inline io/iouring-wrapper.cpp:247]: We used to only submit IO in the reap process, in order to reduce the number of syscalls.

Will it cause performance decline?

File: photon.h [hunk 2]
```diff
@@ -40,17 +43,21 @@ const uint64_t INIT_IO_EXPORTFS = SHIFT(10);
...
+                                    INIT_EVENT_SIGNAL;
+const uint64_t INIT_IO_DEFAULT    = INIT_IO_LIBCURL;
 #endif
 
 #undef SHIFT
 
 struct PhotonOptions {
     int libaio_queue_depth = 32;
+    uint32_t iouring_sq_thread_cpu;
+    uint32_t iouring_sq_thread_idle = 1000;     // by default polls for 1s
```
Comment (beef9999) [inline photon.h:60]: Rename to `iouring_sq_thread_idle_ms` ?

File: io/iouring-wrapper.cpp (hunk 1 above)
Comment (lihuiba) [reply io/iouring-wrapper.cpp:247]: We already have it in async_io() and rm_interest(). It might be missing here?

File: io/iouring-wrapper.cpp [hunk 3]
```diff
@@ -664,18 +698,17 @@ int iouring_unregister_files(int fd) {
...
-    LOG_INFO("Init event engine: iouring ", VALUE(is_master));
-    return NewObj<iouringEngine>(is_master) -> init();
+void* new_iouring_event_engine(iouring_args args) {
+    LOG_INFO("Init event engine: iouring ",
+        make_named_value("is_master",     args.is_master),
+        make_named_value("setup_sqpoll",  args.setup_sqpoll),
+        make_named_value("setup_sq_aff",  args.setup_sq_aff),
+        make_named_value("sq_thread_cpu", args.sq_thread_cpu));
+    auto uring = NewObj<iouringEngine>() -> init(args);
+    if (args.is_master) return uring;
```
Comment (beef9999) [inline io/iouring-wrapper.cpp:708]: uring's type is MasterEventEngine? Don't need to do conversion?

**Review by beef9999 — APPROVED** [review]

File: io/iouring-wrapper.cpp (hunk 3 above)
Comment (lihuiba) [reply io/iouring-wrapper.cpp:708]: no need for the 1st parent class

**Pull request merged**